java/ql/src/Security/CWE/CWE-200/AndroidWebViewSettingsPermitsContentAccess.ql

```diff
@@ -0,0 +1,100 @@
+/**
+ * @name Android WebView settings permits content access
+ * @id java/android/websettings-permit-contentacces
+ * @description Access to content providers in a WebView can permit access to protected information by loading content:// links.
+ * @kind problem
+ * @problem.severity warning
+ * @precision medium
+ * @tags security
+ *      external/cwe/cwe-200
+ */
+
+import java
+import semmle.code.java.dataflow.TaintTracking
+import semmle.code.java.frameworks.android.WebView
+
+/** Represents `android.webkit.WebView` and its subclasses. */
+private class TypeWebViewOrSubclass extends RefType {
+  TypeWebViewOrSubclass() { this.getASupertype*() instanceof TypeWebView }
+}
+
+/**
+ * A method access to a getter method which is private.
+ *
+ * In Kotlin, member accesses are translated to getter methods.
+ */
+private class PrivateGetterMethodAccess extends MethodAccess {
+  PrivateGetterMethodAccess() {
+    this instanceof MethodAccess and
+    this.getMethod() instanceof GetterMethod and
+    this.getMethod().isPrivate()
+  }
+}
+
+/** A source for `android.webkit.WebView` objects. */
+class WebViewSource extends DataFlow::Node {
+  WebViewSource() {
+    this.getType().(RefType) instanceof TypeWebViewOrSubclass and
+    // To reduce duplicate results, we only consider WebView objects from
+    // constructor and method calls, or method accesses which are cast to WebView.
+    (
+      this.asExpr() instanceof ClassInstanceExpr or
+      this.asExpr() instanceof MethodAccess or
+      this.asExpr().(CastExpr).getAChildExpr() instanceof MethodAccess
+    ) and
+    // Avoid duplicate results from Kotlin member accesses.
+    not this.asExpr() instanceof PrivateGetterMethodAccess
+  }
+}
+
+/**
+ * A sink representing a call to `android.webkit.WebSettings.setAllowContentAccess` that
+ * disables content access.
+ */
+class WebSettingsDisallowContentAccessSink extends DataFlow::Node {
+  WebSettingsDisallowContentAccessSink() {
+    exists(MethodAccess ma |
+      ma.getQualifier() = this.asExpr() and
+      ma.getMethod() instanceof AllowContentAccessMethod and
+      ma.getArgument(0).(BooleanLiteral).getBooleanValue() = false
```

Comment:
If we use `CompileTimeConstantExpr` here, we can catch a few more cases that don't directly use a boolean literal at the call site, but rather a constant variable or field:

```suggestion
      ma.getArgument(0).(CompileTimeConstantExpr).getBooleanValue() = false
```